### PR TITLE
Be able to specify source columns in `intersection`

### LINKED ISSUE #148: Be able to specify source columns in `intersection`
It would be great if we could choose those columns we want from polygon dataset.
How difficult would it be @dgaubert?

### PR DESCRIPTION
**Pull request by dgaubert**
Fixes #148

cc @rochoa

### CODE REVIEW

File: lib/node/nodes/intersection.js [hunk 1]
```diff
@@ -1,28 +1,41 @@
...
-    target: Node.PARAM.NODE(Node.GEOMETRY.ANY)
+    source_columns: Node.PARAM.NULLABLE(Node.PARAM.ARRAY(Node.PARAM.STRING()), []),
+    target: Node.PARAM.NODE(Node.GEOMETRY.ANY),
 };
 
-var Intersection = Node.create(TYPE, PARAMS, { cache: true, version: 3 });
+var Intersection = Node.create(TYPE, PARAMS, {
+    cache: true,
+    version: 4,
+    beforeCreate: function (node) {
```
Comment (rochoa) [inline lib/node/nodes/intersection.js]: Why do you need to retrieve the source_colums here?

Comment (dgaubert) [inline lib/node/nodes/intersection.js]: To keep previous behaviour, all columns from source. Is it not what we want?

Comment (rochoa) [inline lib/node/nodes/intersection.js]: But you can do that outside of `beforeCreate`. The `beforeCreate` is supposed to be used when you cannot do something after the node has been created: ignore param for id generation, throw an exception due a specific validation, and similar things.

Comment (dgaubert) [inline lib/node/nodes/intersection.js]: I didn't know that, thanks for clarifying. Fixing ASAP.

Comment (rochoa) [inline lib/node/nodes/intersection.js]: Not a high priority thing, just wanted to be sure it was not actually needed. Also wanted to let you know the intentional behaviour for `beforeCreate` 😉 .

**Pull request merged**